andela/codepirates-ah-backend
# #167313424 Build out user track comments edit history

#### What does this PR do?
-Has tracking comments edit functionality
#### Description of Task to be completed?
- Added functionality for the user to track comments edit history on comment updation
- Tested the comments route
This PR has the following endpoints working 
- UPDATE/http://localhost:8000/api/v1/comments/</id>
#### How should this be manually tested?
- Download or clone the tutorial project code from this repo, in your terminal then git checkout `ft-comments-track-previous-edit-167313424`, install all required NPM packages by running NPM install from the command line in the project root folder (where the package.json is located).
- Start the API by running NPM start from the command line in the project root folder, you should see the message Server listening on port 8000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
- to test authentication and authorization, add this to the header
Key: :x-access-token,value:JWT TOKEN
#### Any background context you want to provide?
 - Implemented the endpoint logic using Sequelize framework 
#### What are the relevant pivotal tracker stories?
[#167313424](https://www.pivotaltracker.com/n/projects/2372121/stories/167313424)
#### Screenshots (if appropriate)

This photo below shows how I get edit history  over the comment, the field called commentRevisions shows the previous comment edited

![image](https://user-images.githubusercontent.com/16412410/63857394-e88e4180-c9a3-11e9-9465-f1ac4b9ddd70.png)

![image](https://user-images.githubusercontent.com/16412410/63857411-f348d680-c9a3-11e9-828c-55ff27787dff.png)

![image](https://user-images.githubusercontent.com/16412410/63857640-5b97b800-c9a4-11e9-9fa9-d4d539de3bbe.png)


Below is the swagger documentation screenshot

![image](https://user-images.githubusercontent.com/16412410/63859335-77e92400-c9a7-11e9-9db1-a05a06498d55.png)


#### Questions:
